File: numberGuess.py
# Number Guessing Game
# Version 0.3
# Simple CLI game to practice loops, conditionals, and input handling

import random

# Generate the target number once at the start
target_number = random.randint(1, 100)
guess_count = 0

print("Guess a number between 1 and 100")

while True:
    user_input = input("Enter your guess: ")

    # Basic input validation to avoid crashes
    if not user_input.isdigit():
        print("Please enter a valid number.")
        continue

    guess = int(user_input)
    guess_count += 1

    if guess > target_number:
        print("Too high!")
    elif guess < target_number:
        print("Too low!")
    else:
        print(f"Congratulations! You guessed the number in {guess_count} attempts.")
        break

File: numberGuess.cpp

// Developed by Gulshan for Google Apprenticeship 2025

#include <iostream>
#include <cstdlib>
#include <ctime>

using namespace std;

int main() {
    srand(time(0));
    int number = rand() % 100 + 1;
    int guess, attempts = 0;

    cout << "Guess a number between 1 and 100\n";

    do {//Loop
        cout << "Enter your guess: ";
        cin >> guess;//Input Handling
        attempts++;

        if (guess > number) {//Conditionals
            cout << "Too high!\n";
        } else if (guess < number) {
            cout << "Too low!\n";
        } else {
            cout << "Congratulations! You guessed it in " << attempts << " tries.\n";
        }
    } while (guess != number);

    return 0;
}